src/spec.h

```diff
@@ -349,18 +349,13 @@ int CompareVestions(Version v1, Version v2);
  * Retrieves the current spec cache from the index, incrementing its
  * reference count by 1. Use IndexSpecCache_Decref to free
  */
-IndexSpecCache *IndexSpec_GetSpecCache(const IndexSpec *spec);
+const IndexSpecCache *IndexSpec_GetSpecCache(const IndexSpec *spec);
 
 /**
  * Decrement the reference count of the spec cache. Should be matched
  * with a previous call of GetSpecCache()
  */
-void IndexSpecCache_Decref(IndexSpecCache *cache);
-
-/**
- * Create a new copy of the spec cache from the current index spec
- */
-IndexSpecCache *IndexSpec_BuildSpecCache(const IndexSpec *spec);
+void IndexSpecCache_Decref(const IndexSpecCache *cache);
```

Comment:
this is not const

Reply:
the cache reference might be const (ex. in an RLookup) so this way we can decrement the reference without getting warnings